packages/server/src/api/controllers/application.ts

```diff
@@ -502,22 +504,24 @@ export const sync = async (ctx: any, next: any) => {
 }
 
 const updateAppPackage = async (appPackage: any, appId: any) => {
-  const db = getAppDB({ appId })
-  const application = await db.get(DocumentTypes.APP_METADATA)
+  return doInAppContext(appId, async () => {
+    const db = getAppDB()
```

Comment:
What was the bug related to this out of curiosity?

Reply:
The app context in the overview screen is set to the dev app, so when the update request is made there it modifies the dev app icon. For published applications, this gives the impression that no change was made.
This change ensures the same update behaviour as the app list. When updating the icon in the app list, the context is resolved to the prod application if the app is published.